mobile/v1/snowflake/sql-runner/sql/standard/00-setup/01-main/01-stored-procedures.sql

```diff
@@ -0,0 +1,245 @@
+/*
+   Copyright 2021 Snowplow Analytics Ltd. All rights reserved.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+
+/*
+  MK_TRANSACTION
+  Side effects procedure.
+  Input: a concatenation of one or more DML sql statements split by semicolon.
+  It is important that the statements are not DDL.
+  Either all the statements in the block succeed and get committed or all get rolled back.
+  To drop:
+  DROP PROCEDURE {{.output_schema}}.mk_transaction(VARCHAR);
+*/
+CREATE OR REPLACE PROCEDURE {{.output_schema}}.mk_transaction(DML_STATEMENTS VARCHAR)
+  RETURNS VARCHAR
+  LANGUAGE JAVASCRIPT
+  AS
+  $$
+
+  dmls = DML_STATEMENTS.split(';')
```

Comment:
Is there a reason to split the statements? I'm not seeing any benefit and it intuitively is better to just execute the SQL we are provided rather than arbitrarily splitting it into separate statements. 

If there are a set of related statements which include the delimiter `;`, then `mk_transaction()` won't be able to run them.

Not a blocker for this release, but if there isn't a reason to do it we should look to remove this in future.

Reply:
I believe we have to be split as Snowflake does not support running multiple sql statements via `execute()`.